pkg/platform/model/vcs.go

```diff
@@ -235,13 +235,20 @@ func CommitsBehindLatest(ownerName, projectName, commitID string) (int, *failure
 type Changeset = []*mono_models.CommitChangeEditable
 
 // AddChangeset creates a new commit with multiple changes as provided. This is lower level than CommitChangeset.
-func AddChangeset(parentCommitID strfmt.UUID, commitMessage string, changeset Changeset) (*mono_models.Commit, *failures.Failure) {
+func AddChangeset(parentCommitID strfmt.UUID, commitMessage string, isHeadless bool, changeset Changeset) (*mono_models.Commit, *failures.Failure) {
 	params := vcsClient.NewAddCommitParams()
-	params.SetCommit(&mono_models.CommitEditable{
+
+	commit := &mono_models.CommitEditable{
 		Changeset:      changeset,
 		Message:        commitMessage,
 		ParentCommitID: parentCommitID,
-	})
+	}
+
+	if isHeadless {
+		commit.AnonID = logging.UniqID()
+	}
+
+	params.SetCommit(commit)
```

Comment:
This is too much specialized knowledge that we shouldn't be dealing with at this level. Instead of passing `isHeadless` just pass the anonymous ID.